### PR TITLE
Issue 40347: UX Domain designer wrappers not supporting domain kind specific properties in the create case

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Issue [40347](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=40347): Domain designer wrappers not supporting domain kind specific properties in the create case. Allow for calls to Domain.getDomainDetails to pass in a domainKind param to get domain kind specific properties. This allows for the create case for each domain designer to use the domain kind specific options in the DomainForm components (i.e. fields panel).

#### Related Pull Requests
* https://github.com/LabKey/labkey-api-js/pull/62
* https://github.com/LabKey/labkey-ui-components/pull/248

#### Changes
* Add DomainUtil.getTemplateDomainForDomainKind() to get the domain kind specific properties for a temp GWTDomain object
* Add support for domainKind prop in PropertyController.GetDomainDetailsAction
* Update domain designer wrappers to get model response for the create case based on properties from domain kind
* Update to @labkey/components package with related changes

### CODE REVIEW

File: list/src/client/Designer/Designer.tsx [hunk 1]
```diff
@@ -22,7 +22,6 @@ import {
     ListDesignerPanels,
     ListModel,
     fetchListDesign,
```
Comment (labkey-nicka) [inline list/src/client/Designer/Designer.tsx:24]: nit: In the issues variant of `Designer.tsx` there are unused imports.

File: experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx [hunk 2]
```diff
@@ -73,6 +66,8 @@ export class App extends React.PureComponent<any, State> {
         // if the URL has a RowId param, look up the sample type info for the edit case
         // else we are in the create new sample type case
         const rowId = this.getRowIdParam();
+        const { name, nameReadOnly } = ActionURL.getParameters();
```
Comment (labkey-ankurj) [inline experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx]: could be moved in else?

**Review by labkey-ankurj — APPROVED** [review]

File: list/src/client/Designer/Designer.tsx (hunk 1 above)
Comment (cnathe) [reply list/src/client/Designer/Designer.tsx:24]: Unused imports removed in a different, recently merged PR: https://github.com/LabKey/platform/pull/1184

File: experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx (hunk 2 above)
Comment (cnathe) [reply experiment/src/client/SampleTypeDesigner/SampleTypeDesigner.tsx]: Yes, done.

**Pull request merged**